src/Quaternion.jl

```diff
@@ -44,7 +44,11 @@ end
 
 real(::Type{Quaternion{T}}) where {T} = T
 real(q::Quaternion) = q.s
-imag(q::Quaternion) = [q.v1, q.v2, q.v3]
+imag_part(q::Quaternion) = (q.v1, q.v2, q.v3)
+function imag(q::Quaternion)
+    Base.depwarn("`imag(q::Quaternion)` is deprecated, use `collect(imag_part(q))` instead.", :imag)
+    collect(imag_part(q))
+end
```

Comment:
You can use `@deprecate` without exporting `imag` like this:
```suggestion
@deprecate imag(q::Quaternion) collect(imag_part(q)) false
```

Reply:
Oh, I didn't know the usage. thanks!